In [6]:
# Import all necessary libraries:
import pandas as pd 
import numpy as np 
import scipy as scp
import sklearn
import statsmodels.api as sm  

from imblearn.over_sampling import SMOTE 
from sklearn.model_selection import train_test_split

import warnings  
warnings.filterwarnings('ignore')  

df = pd.read_csv('LSE_DA301_Week_2_files/Data/ecommerce.csv')

df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 506 entries, 0 to 505
Data columns (total 12 columns):
 #   Column     Non-Null Count  Dtype  
---  ------     --------------  -----  
 0   Sale       506 non-null    float64
 1   por_OS     506 non-null    float64
 2   por_NON    506 non-null    float64
 3   recc       506 non-null    int64  
 4   avg_no_it  506 non-null    float64
 5   age        506 non-null    float64
 6   dis        506 non-null    float64
 7   diff_reg   506 non-null    int64  
 8   tax        506 non-null    int64  
 9   bk         506 non-null    float64
 10  lowstat    506 non-null    float64
 11  Median_s   506 non-null    float64
dtypes: float64(9), int64(3)
memory usage: 47.6 KB


In [7]:
df.head()

Sale  por_OS  por_NON  recc  avg_no_it   age     dis  diff_reg  tax  \
0  0.63    18.0     2.31     0      6.575  65.2  4.0900         1  296   
1  2.73     0.0     7.07     0      6.421  78.9  4.9671         2  242   
2  2.73     0.0     7.07     0      7.185  61.1  4.9671         2  242   
3  3.24     0.0     2.18     0      6.998  45.8  6.0622         3  222   
4  6.91     0.0     2.18     0      7.147  54.2  6.0622         3  222   

       bk  lowstat  Median_s  
0  396.90     4.98      24.0  
1  396.90     9.14      21.6  
2  392.83     4.03      34.7  
3  394.63     2.94      33.4  
4  396.90     5.33      36.2

In [10]:
# Specify that the column Median_s 
# should be moved into a separate DataFrame.
cols = df.columns[df.columns != 'Median_s']  

# Specify ‘X’ as the independent variables 
# and ‘y’ as the dependent variable:
X = df[cols]
y = df['Median_s']

# Split the data training and testing 30/70:
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.3, random_state=0)

# Import the ‘DecisionTreeRegressor’ class from sklearn.
from sklearn.tree import DecisionTreeRegressor  

# Create the ‘DecisionTreeRegressor’ class 
# (which has many parameters; input only #random_state=0):
regressor = DecisionTreeRegressor(random_state=0)

# Fit the regressor object to the data set.
regressor.fit(X_train,y_train)

DecisionTreeRegressor(random_state=0)

In [11]:
# Import the necessary packages:
from sklearn import metrics
import math

# Predict the response for the data test.
y_predict = regressor.predict(X_test)  

# Specify to print the MAE and MSE (to evaluate the accuracy of the new model):
print("Mean Absolute Error: ", metrics.mean_absolute_error(y_test, y_predict))
print("Mean Squared Error: ", metrics.mean_squared_error(y_test, y_predict))
# [3b] Calculate the RMSE.
print("Root Mean Squared Error: ", 
     math.sqrt(metrics.mean_squared_error(y_test, y_predict)))  

Mean Absolute Error:  3.2552631578947366
Mean Squared Error:  26.397894736842108
Root Mean Squared Error:  5.137888159238395



RMSE - MAE = 5.14 - 3.26 = 1.88
1.88 is considered a small number since it is close to 0, i.e. no large errors in the forecast

